File: Front/src/app/pc-laptop/computers.ts
import {ProductType} from "../shopping-cart/shopping-cart-item";

export interface Computers {
  idKomputera: number;
  model: string;
  system: string;
  procesor: string;
  kgraficzna: string;
  pamRam: string;
  cena: number;
  img: string;
  productType: ProductType;
}

File: Back/src/main/java/com/api/back/model/Computers.java
package com.api.back.model;

import com.api.back.Service.ShoppingCart.Enum.ProductType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Computers implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long idKomputera;
    private String model;
    private String system;
    private String procesor;
    private String kGraficzna;
    private String pamRam;
    private int cena;
    private String img;
    @Enumerated(EnumType.STRING)
    private ProductType productType;
}

